src/containers/setContacts/ducks/types.ts

```diff
@@ -0,0 +1,85 @@
+import { Moment } from 'moment';
+import { ThunkAction } from 'redux-thunk';
+import { ProtectedApiExtraArgs } from '../../../api/protectedApiClient';
+import { C4CState } from '../../../store';
+import { AsyncRequest } from '../../../utils/asyncRequest';
+import { FileField } from '../../../utils/fileEncoding';
+import { ContactsActions } from './actions';
+
+export interface ContactsReducerState {
+  readonly contacts: AsyncRequest<ContactInfo, any>;
+  readonly setContacts: AsyncRequest<void, any>;
+}
+
+export type ContactsThunkAction<R> = ThunkAction<
+  R,
+  C4CState,
+  ProtectedApiExtraArgs,
+  ContactsActions
+>;
+
+interface GenericContact<D, P> {
+  email: string;
+  firstName: string;
+  lastName: string;
+  phoneNumber: string;
+  dateOfBirth: D;
+  pronouns?: string;
+  allergies?: string;
+  diagnosis?: string;
+  medications?: string;
+  notes?: string;
+  photoRelease?: boolean;
+  referrer?: string;
+  profilePicture?: P;
+}
+
+interface GenericChild<D, P> {
+  firstName: string;
+  lastName: string;
+  pronouns?: string;
+  school: string;
+  schoolYear: string;
+  allergies?: string;
+  diagnosis?: string;
+  medications?: string;
+  notes?: string;
+  photoRelease?: boolean;
+  profilePicture?: P;
+  dateOfBirth: D;
+}
```

Comment:
Lots of redundant fields. Consider using `Omit` https://www.typescriptlang.org/docs/handbook/utility-types.html#omittype-keys for creating GenericChild from GenericContact, omit the properties you don't need, and then simply add the ones that are new.

Reply:
Good idea, I think we can avoid that for simplicity and make a AbstractContactInfo or something that both contacts and child extend

Reply:
I agree, abstract is good